File: android/NotifyBridge/app/src/main/java/com/alpware/notifybridge/network/NotificationActionCommand.kt
package com.alpware.notifybridge.network

import com.google.gson.annotations.SerializedName

/**
 * Represents a notification action requested by the paired Mac device.
 */
data class NotificationActionCommand(
    @SerializedName("id")
    val id: String,

    @SerializedName("type")
    val type: String,

    @SerializedName("notificationKey")
    val notificationKey: String?,

    @SerializedName("packageName")
    val packageName: String?,

    @SerializedName("replyText")
    val replyText: String?,

    @SerializedName("createdAt")
    val createdAt: String?
)

File: macos/NotifyBridgeMac/NotifyBridgeMac/NotifyBridgeMac/NotificationActionCommand.swift
//
//  NotificationActionCommand.swift
//  NotifyBridgeMac
//
//  Created by Mahmut Alperen Ünal on 10.06.2026.
//

import Foundation

/// Represents a notification action queued by the Mac app for execution on Android.
struct NotificationActionCommand: Codable, Identifiable {
    let id: String
    let type: NotificationActionType
    let notificationKey: String?
    let packageName: String?
    let replyText: String?
    let createdAt: String
}

/// Supported notification actions that can be executed on the Android device.
enum NotificationActionType: String, Codable {
    case dismiss
    case openOnPhone
    case reply
}
